File: ProgrammersLifeiOS/ProgrammersLifeiOS/SkillRequirement.m
//
//  SkillRequirement.m
//  ProgrammersLifeiOS
//
//  Created by Zackary T Misso on 7/6/15.
//  Copyright © 2015 Zackary Misso. All rights reserved.
//

#import "SkillRequirement.h"
#import "Skill.h"

@implementation SkillRequirement

-(id)init{
    if(self = [super init]){
        skill = NULL;
        neededLevel = -1;
    }
    return self;
}

-(BOOL)checkRequirement{
    if([skill getLevel] >= neededLevel)
        return YES;
    return NO;
}

// getter methods
-(Skill*)getSkill { return skill; }
-(int)getNeededLevel { return neededLevel; }

// setter methods
-(void)setSkill:(Skill*)value { skill = value; }
-(void)setNeededLevel:(int)value { neededLevel = value; }

@end

File: ProgrammersLifeDemo/src/dataStructures/SkillRequirement.java
// This is the Game Tester's version of ProgrammersLife
// Author :: Zackary Misso / FiralStudios
// Version :: 0.0.1

package dataStructures;

public class SkillRequirement {
	private Skill skill;
	private int neededLevel;
	private int forLevel;
	
	public boolean checkReq(){
		if(skill.getLevel() >= neededLevel)
			return true;
		return false;
	}
	
	public boolean checkReqForLevel(int lev){
		if(lev > forLevel)
			return true;
		return checkReq();
	}
	
	// getter methods
	public Skill getSkill(){return skill;}
	public int getNeededLevel(){return neededLevel;}
	public int getForLevel(){return forLevel;}
	
	// setter methods
	public void setSkill(Skill param){skill = param;}
	public void setNeededLevel(int param){neededLevel = param;}
	public void setForLevel(int param){forLevel = param;}
}
